pkg/reconciler/taskrun/resources/validate_params.go

```diff
@@ -45,3 +46,40 @@ func ValidateOutOfBoundArrayParams(declarations v1.ParamSpecs, params v1.Params,
 	}
 	return nil
 }
+
+func validateStepHasStepActionParameters(ctx context.Context, stepParams v1.Params, stepActionDefaults []v1.ParamSpec) error {
```

Comment:
looks like the ctx is not used?

Reply:
I wonder if we have the same logic from Task? Since we are just passing v1.Params and v1.ParamSpec and they are not specific for Step. I wonder if we could reuse some code. But I cannot find it atm

Reply:
Oh might be this one: https://github.com/tektoncd/pipeline/blob/a106110878496a25e4a863b19cd8f13a7369843c/pkg/reconciler/taskrun/validate_taskrun.go#L36

Do you think we can reuse it?

Reply:
Not quite because in the function `validateStepHasStepActionParameters ` we cannot validate the types and properties yet since the parameter replacements have not yet happened. Here, we want to ensure that the Step is passing params to the StepAction. The exactly value and type of the param passed by Step is still unknown since it is likely something like `$(params.foo)` instead of the resolved value.